hpx/lcos/detail/async_implementations.hpp

```diff
@@ -148,11 +149,11 @@ namespace hpx { namespace detail
 
         if (target_is_managed)
         {
-            typedef typename lcos::detail::shared_state_ptr_for<
+            typedef typename traits::detail::shared_state_ptr_for<
                 future<result_type>
             >::type shared_state_ptr;
 
-            shared_state_ptr const& state = lcos::detail::get_shared_state(f);
+            shared_state_ptr const& state = traits::get_shared_state(f);
```

Comment:
This looks odd. The trait is defined in `hpx::traits` but the returned type is defined in `hpx::traits::detail`.